lib/actions/racking.rb

```diff
@@ -10,82 +10,76 @@ class InvalidDataParams < StandardError
   def initialize(message = nil)
     super(message)
     @errors = message
-    #@msg = html_error_message([message].flatten)
   end
 
   def html_error_message(error_messages)
     ['<ul>', error_messages.map do |msg|
-      ['<li>',msg,'</li>']
-    end, '</ul>'].flatten.join('')
+      ['<li>', msg, '</li>']
+    end, '</ul>'].flatten.join
   end
-
 end
 
 module Actions
   module Racking
-
     DNA_STOCK_PLATE_PURPOSE = 'DNA Stock Plate'
     RNA_STOCK_PLATE_PURPOSE = 'RNA Stock Plate'
     STOCK_PLATE_PURPOSE = 'Stock Plate'
     DNA_ALIQUOT = 'DNA'
     RNA_ALIQUOT = 'RNA'
-    TUBE_TO_PLATE_TRANSFERRABLE_PROPERTIES = [:study_name,:aliquotType]
+    TUBE_TO_PLATE_TRANSFERRABLE_PROPERTIES = [:study_name, :aliquotType]
+
+    ALIQUOT_PURPOSE = {
+      DNA_ALIQUOT => DNA_STOCK_PLATE_PURPOSE,
+      RNA_ALIQUOT => RNA_STOCK_PLATE_PURPOSE
+    }
 
     # Actions
-    def rack_layout(asset_group)
-      content = selected_file(asset_group).data
-      csv_parsing(asset_group, Parsers::CsvLayout::CsvParser.new(content))
+    def rack_layout(options = {})
```

Comment:
Asset group is an instance variable, so doesn't need to be passed around as. alocal variable.